src/package.ts

```diff
@@ -0,0 +1,57 @@
+import { privateEncrypt } from 'crypto';
+import contextKeys from './contextKeys';
+import { exec, pathExists } from './utilities';
+
+export class SPMPackage {
+	private constructor(
+        readonly folder: string,
+        public contents: any
+    ) {
+        this.setContextKeys()
+    }
+
+    public static async create(folder: string): Promise<SPMPackage> {
+        try {
+            let contents = await SPMPackage.loadPackage(folder)
+            return new SPMPackage(folder, contents)
+        } catch(error) {
+            // TODO: output errors
+            return new SPMPackage(folder, null)
+        }
+    }
+
+    public static async loadPackage(folder: string): Promise<any> {
+        const { stdout } = await exec('swift package describe --type json', { cwd: folder });
+        return JSON.parse(stdout)
+    }
+
+    public async reload() {
+        try {
+            this.contents = await SPMPackage.loadPackage(this.folder)
+        } catch {
+            
+        }
+        this.setContextKeys()
+    }
+
+    get products(): Array<any> {
```

Comment:
This is, for example, where we could benefit from using interfaces to declare the fields we expect to see in the JSON output.

Reply:
ok will set them up

Reply:
done